File: cubes.py
#this is for the finding the cube is perfect or not

class Solution:
    def my_func(self, n):  
        if n < 0:
            n = abs(n)

        root = round(n**(1/3))
        return root**3 == n

# Example usage
if __name__ == "__main__":
    solution = Solution()
    param1 = 8
    result = solution.my_func(param1)

    print(result)

File: Cubes.java
public class Cubes {
    public boolean myFunc(int n) {
        
        long val = Math.abs((long) n);

        long root = Math.round(Math.pow(val, 1.0 / 3.0));

        return (root * root * root) == val;
    }

    // main function
    public static void main(String[] args) {
        Solution solution = new Solution();
        int param1 = 8;
        boolean result = solution.myFunc(param1);

        System.out.println(result);
    }
}
